minor: build failed on Arch

Hi,

I got the following build failure on Arch:
```
…/hpp-constraints-4.6.0/src/generic-transformation.cc:162:76: error: no matching function for call to 'liegroupSpace<hpp::constraints::GenericTransformation<6>::ComputePosition, hpp::constraints::GenericTransformation<6>::ComputeOrientation, hpp::constraints::GenericTransformation<6>::OutputSE3>(std::vector<bool>&)'
…/hpp-constraints-4.6.0/src/generic-transformation.cc:166:76: error: no matching function for call to 'indices<hpp::constraints::GenericTransformation<6>::ComputePosition, hpp::constraints::GenericTransformation<6>::ComputeOrientation, hpp::constraints::GenericTransformation<6>::OutputSE3>(std::vector<bool>&)'
```

GCC 9.1.0
Eigen 3.3.7
Boost 1.69.0

As this doesn't happen on 14.04, 16.04 & 18.04, this bug report is only a notice, which can be dismissed at will.

--- a/src/generic-transformation.cc
+++ b/src/generic-transformation.cc
@@ -159,11 +159,11 @@ namespace hpp {
       (const std::string& name, const DevicePtr_t& robot,
        std::vector <bool> mask) :
         DifferentiableFunction (robot->configSize (), robot->numberDof (),
-            liegroupSpace <ComputePosition, ComputeOrientation, OutputSE3> (mask),
+            liegroupSpace <(bool)ComputePosition, (bool)ComputeOrientation, (bool)OutputSE3> (mask),
             name),
         robot_ (robot),
         m_(robot->numberDof()-robot->extraConfigSpace().dimension()),
-        Vindices_ (indices<ComputePosition, ComputeOrientation, OutputSE3> (mask)),
+        Vindices_ (indices<(bool)ComputePosition, (bool)ComputeOrientation, (bool)OutputSE3> (mask)),
         mask_ (mask)
     {
       assert(mask.size()==DerSize);

--- a/tests/convex-shape.cc
+++ b/tests/convex-shape.cc
@@ -15,6 +15,9 @@
 // hpp-constraints. If not, see <http://www.gnu.org/licenses/>.
 
 #define BOOST_TEST_MODULE ConvexShape
+
+#include <pinocchio/fwd.hpp>
+
 #include <boost/test/included/unit_test.hpp>
 
 #include "hpp/constraints/convex-shape.hh"

--- a/tests/multithread.cc
+++ b/tests/multithread.cc
@@ -15,6 +15,9 @@
 // hpp-constraints. If not, see <http://www.gnu.org/licenses/>.
 
 #define BOOST_TEST_MODULE multithread_test
+
+#include <pinocchio/fwd.hpp>
+
 #include <boost/test/included/unit_test.hpp>
 
 #include <stdlib.h>

--- a/tests/symbolic-calculus.cc
+++ b/tests/symbolic-calculus.cc
@@ -15,6 +15,9 @@
 // hpp-constraints. If not, see <http://www.gnu.org/licenses/>.
 
 #define BOOST_TEST_MODULE SymbolicCalculus
+
+#include <pinocchio/fwd.hpp>
+
 #include <boost/test/included/unit_test.hpp>
 
 #include <stdlib.h>

